### PR TITLE
setExecutionSpace thread safety

### PR DESCRIPTION
**Pull request by neelakausik**
Made m_current_execution_space and m_synced_since_last_kernel thread_local to address thread safety concerns.

### CODE REVIEW

File: src/chai/ArrayManager.cpp [hunk 1]
```diff
@@ -32,7 +34,7 @@ ArrayManager::ArrayManager() :
   m_callbacks_active{true}
 {
   m_pointer_map.clear();
-  m_current_execution_space = NONE;
+  ArrayManager::m_current_execution_space = NONE;
```
Comment (adayton1) [inline src/chai/ArrayManager.cpp:37]: Adding ArrayManager:: here shouldn't be necessary.

**Review by adayton1 — DISMISSED** [review]
LGTM

File: src/chai/ArrayManager.cpp (hunk 1 above)
Comment (adayton1) [reply src/chai/ArrayManager.cpp:37]: Or inside any ArrayManager methods.

**Review by adayton1 — APPROVED** [review]
LGTM

**Comment by dtaller** [comment]
Looks good to me.

Was there a failing test or a something like that that prompted this change, or were you just inspecting the code for possible issues?

**Review by dtaller — APPROVED** [review]

**Comment by adayton1** [comment]
> Looks good to me.
> 
> Was there a failing test or a something like that that prompted this change, or were you just inspecting the code for possible issues?

@dtaller, we've been looking at launching kernels in separate GPU streams from separate CPU threads, which has led us to look into thread safety in CHAI and RAJA. This was one of the main issues that was uncovered.

**Comment by davidbeckingsale** [comment]
Merged in #230

**Pull request closed without merging**